management/http.go

```diff
@@ -62,12 +62,12 @@ func (client *Client) SendAzurePutRequest(url string, contentType string, data [
 	}
 
 	requestId := response.Header[requestIdHeader]
-	return requestId[0], nil
+	return RequestId(requestId[0]), nil
```

Comment:
actually this could panic if [0] doesn't exist. can we do a len check here and return just err if even header isn't present?